specs/SharedBuffer.md

```diff
@@ -0,0 +1,333 @@
+  further processed or fed to other parts of the app or OS. For example, the web page
+  generates 100MBs of high DPI images to be printed and needs to pass that to the native
+  code to print. See https://github.com/MicrosoftEdge/WebView2Feedback/issues/89.
+- Native side generates a large amount of data for the web side to consume. The data
+  might or might not come directly from files. For example the native side has generated
+  terrabytes of data to produce different graphs on the web side.
+  See https://github.com/MicrosoftEdge/WebView2Feedback/issues/1005.
+
+To support these scenarios, we are adding an Edge WebView2 API to support sharing
+buffers between the WebView2 host app process and WebView2 renderer process, based
+on shared memory from the OS.
+
+The application code can use the APIs to create a shared buffer object, and share to
+scripts as [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) object.
+Then both the native application code and the script will be able to access the same memory.
+
+# Conceptual pages (How To)
+
+Besides using the memory address directly, the shared buffer object can be accessed
+from native side via an IStream* object that you can get from the shared buffer object.
+
+When the application code calls `PostSharedBufferToScript`, the script side will
+receive a `SharedBufferReceived` event containing the buffer as an `ArrayBuffer` object.
+After receiving the shared buffer object, it can access it the same way as any other
+ArrayBuffer object, including transering to a web worker to process the data on
+the worker thread.
+
+As shared buffer normally represent a large memory, instead of waiting for garbage
+collection to release the memory along with the owning objects, the application
+should try to release the buffer as soon as it doesn't need access to it.
+This can be done by calling Close() method on the shared buffer object, or
+`chrome.webview.releaseBuffer` from script.
+
+As the memory could contain sensitive information and corrupted memory could crash
+the application, the application should only share buffer with trusted sites.
+
+The application can use other messaging channel like `PostWebMessageAsJson` and
+`chrome.webview.postMessage` to inform the other side the desire to have a shared
+buffer and the status of the buffer (like data is produced or consumed).
+
+# Examples
+
+The example below illustrates how to send data from application to script for one time read only consumption.
+
+The script code will look like this:
+```
+        window.onload = function () {
+            window.chrome.webview.addEventListener("SharedBufferReceived", e => {
+                SharedBufferReceived(e);
+            });
+        }
+
+        function SharedBufferReceived(e) {
+            if (e.data && e.data.readOnly) {
+                // This is the one time read only buffer
+                let oneTimeSharedBuffer = e.sharedBuffer;
```

Comment:
Why is this called "one time" shared buffer? There's nothing the app did to make this a one-time buffer, so I guess this is saying that "one time"-ness is coming from the system somehow? What is a "one time" buffer anyway? Can I create only one view from it? How can I make it a "multi-use" buffer?

Reply:
Will rename it as contosoDisplayBuffer, to be aligned with the other updates to the sample code.

The one-time here has no deep meaning. I was trying to say that the app only uses the buffer to send data once, and neither side want to access it after sending the data over once. The same buffer object can be used to send different multiple times. One side just has to notify the other side that data is updated.